packages/docutils/lib/cli/check.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * Provides functions to validate user-provided options as part of Yargs' post-parsing "check" phase
+ * @module
+ */
+
+import {fs} from '@appium/support';
+import _ from 'lodash';
+import pluralize from 'pluralize';
+import {InferredOptionTypes, Options} from 'yargs';
+import logger from '../logger';
+
+const log = logger.withTag('check');
+
+/**
+ * Given a list of objects with `id` and `path` props, filters out the ones that do not exist
+ * @param paths Filepaths
+ * @returns Missing files
+ */
+async function filterMissing(paths: MissingFileData[]): Promise<MissingFileData[]> {
+  const exists = await Promise.all(
+    paths.map(async ({id, path}) => ({id, path, exists: await fs.exists(path)}))
+  );
+  const results = _.reject(exists, 'exists');
+  return _.map(results, (result) => _.omit(result, 'exists'));
+}
+
+/**
+ * Data structure describing a missing file; returned by {@linkcode filterMissing}
+ */
+interface MissingFileData {
+  /**
+   * Arbitrary identifier; intent was to map to an option name
+   */
+  id: string;
+  /**
+   * Normalized filepath
+   */
+  path: string;
+}
+
+/**
+ * Takes user-provided paths and checks for existence.
+ *
+ * Filters options to consider based on group name only.
+ *
+ * Meant to be used as a "fail-fast" strategy on the CLI, so we don't go all the way through some
+ * expensive behavior before realizing we're missing a path.
+ * @param opts Options object for a yargs command
+ * @param group Group name to filter on
+ * @param argv User-provided args
+ * @returns `true` if all paths exist or otherwise an error message
```

Comment:
would it make more sence to just throw an error instead of returning a string?

Reply:
we could do that, but I figure that'd produce a stack trace somewhere, and it's unneeded for checking parameters.  the return value of these functions is just what yargs expects.